### PR TITLE
react-native app: Assignment Day 11

### PR DESCRIPTION
**Pull request by Mustafalightwala**
An app with Bottom tab navigation having 2 tabs:
- On the first tab- User is able to open the camera and gallery to select an image.
- On the second tab- A button which when user clicks will see the loader for 3 seconds then the loader automatically closes.
Screenshot_1613484041
Screenshot_1613484066
Screenshot_1613484068
Screenshot_1613484075
Screenshot_1613484078
Screenshot_1613484083
Screenshot_1613484087
Screenshot_1613484092

### CODE REVIEW

File: assignment_day_11/App.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import React from 'react';
+import TabNavigator from './src/navigations/MainNavigation';
+
+const App = () => {
```
Comment (vinay-deqode) [inline assignment_day_11/App.js]: Cleaner way:
```
const App = () => <TabNavigator />;
```

File: assignment_day_11/src/components/Button.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
+import React, {Component} from 'react';
+import {View, Text, StyleSheet, TouchableOpacity} from 'react-native';
+import colors from '../constants/colors';
+
+class Button extends Component {
```
Comment (vinay-deqode) [inline assignment_day_11/src/components/Button.js]: we can convert this into a functional component. why you created this as class ?

File: assignment_day_11/src/components/Dialog.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+        </View>
+      </View>
+    </Modal>
+  );
+};
+
+const styles = StyleSheet.create({
+  outerDialog: {
+    flex: 1,
+    backgroundColor: '#00000020',
```
Comment (vinay-deqode) [inline assignment_day_11/src/components/Dialog.js]: you already created constants for colors? don't you think you should refer to the constant file

File: assignment_day_11/src/constants/styles.js [hunk 4]
```diff
@@ -0,0 +1,17 @@
+import {StyleSheet} from 'react-native';
+
+const styles = StyleSheet.create({
```
Comment (vinay-deqode) [inline assignment_day_11/src/constants/styles.js:3]: Do not create separate files for styles. Look into the wrapper container component. Those are also used for the same purpose

File: assignment_day_11/src/navigations/MainNavigation.js [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+          if (route.name === 'Image') {
+            iconName = focused ? 'add-circle' : 'add-circle-outline';
+          } else if (route.name === 'Loading') {
+            iconName = focused ? 'reload-circle' : 'reload-circle-outline';
+          }
+          return <Icon name={iconName} size={size} color={color} />;
+        },
+      })}
+      tabBarOptions={{
+        labelStyle: {
```
Comment (vinay-deqode) [inline assignment_day_11/src/navigations/MainNavigation.js]: avoid inline style

**Review by vinay-deqode — COMMENTED** [review]
Overall looks good. some points
- Permission management should be modular that could use other places too.
- Code needs more cosistency

File: assignment_day_11/App.js (hunk 1 above)
Comment (Mustafalightwala) [reply assignment_day_11/App.js]: Will use this from now.

File: assignment_day_11/src/components/Button.js (hunk 2 above)
Comment (Mustafalightwala) [reply assignment_day_11/src/components/Button.js]: Just for practice, I created one component with the functional approach and one with the class approach.
Will make sure it is not repeated in the future

File: assignment_day_11/src/components/Dialog.js (hunk 3 above)
Comment (Mustafalightwala) [reply assignment_day_11/src/components/Dialog.js]: Apologies did it in a hurry at that time.
Forgot to add it to the constants file

File: assignment_day_11/src/constants/styles.js (hunk 4 above)
Comment (Mustafalightwala) [reply assignment_day_11/src/constants/styles.js:3]: Ok sir
Will look into this

File: assignment_day_11/src/navigations/MainNavigation.js (hunk 5 above)
Comment (Mustafalightwala) [reply assignment_day_11/src/navigations/MainNavigation.js]: ok sir
Will avoid doing this from now

**Review by vinay-deqode — APPROVED** [review]

**Pull request merged**